# Algorytmy losowe
r

In [1]:
import random

random.seed(0)

random.random()

0.8444218515250481

Używając standardowych bilbiotek naszych ulubionych języków programowania, możemy uzyskać liczby losowe. Przykładowo Python oferuje moduł `random`, który zawiera funkcje do generowania liczb losowych. Choć dla nas jako konsumentów bibliotek jest to niewidoczne, wybór odpowiedniego algorytmu jest bardzo istotny. Jakie są ważne metryki takich algorytmów?

## Budowa generatora liczb losowych

Generator liczb losowych to funkcja, która przyjmuje pewien stan (nazywany *ziarnem*, *seed*) i zwraca kolejną liczbę losową oraz nowy stan. Dla określonego ziarna generator zawsze zwraca tę samą sekwencję liczb losowych, dlatego powyższy blok kodu zawsze zwróci nam `0.8444218515250481`. 